### PR TITLE
Implement in-memory caching for Kronos

### PR DESCRIPTION
**Pull request by miteshvp**

### CODE REVIEW

File: analytics_platform/kronos/deployment/rest_api.py [hunk 1]
```diff
@@ -116,22 +120,32 @@ def train_and_save_kronos():
 def predict_and_score():
     input_json = request.get_json()
 
-    app.logger.info("Analyzing the given EPV")
-    app.logger.info(input_json)
+    # Get the response if already cached earlier
+    # Sort package_list elements first
+    for i in range(0, len(input_json)):
```
Comment (rootAvish) [inline analytics_platform/kronos/deployment/rest_api.py]: Minor nitpick, I would prefer if this was done using `enumerate`.

File: analytics_platform/kronos/deployment/rest_api.py [hunk 2]
```diff
@@ -116,22 +120,32 @@ def train_and_save_kronos():
...
+        input_json[i]['package_list'] = sorted(input_json[i]['package_list'])
+
+    hash_key = hashlib.sha224(json.dumps(input_json, sort_keys=True).encode('utf-8')).hexdigest()
```
Comment (rootAvish) [inline analytics_platform/kronos/deployment/rest_api.py]: Any particular reason for using sha224 or did we just pick a random hashing function?

File: analytics_platform/kronos/deployment/rest_api.py (hunk 1 above)
Comment (rootAvish) [reply analytics_platform/kronos/deployment/rest_api.py]: Or in case this json will always be a list- ```python for i in input_json:```

**Comment by miteshvp** [comment]
@rootAvish  - we are using random sha for this purpose.

**Comment by rootAvish** [comment]
[test]

**Comment by miteshvp** [comment]
[test]

**Comment by rootAvish** [comment]
[test]

**Comment by rootAvish** [comment]
LGTM.

**Pull request merged**